### PR TITLE
David user

### PR DESCRIPTION
**Pull request by DavidTTran**
Completed
- User stories 1, 10-12
- Created `profiles_controller`, `users_controller`, `register_controller`
- Added edge cases for creating a user without necessary information
- SimpleCov returns 100%

### CODE REVIEW

File: app/controllers/register_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      render :new
+    elsif params[:confirm_pass] != params[:password]
+      flash[:error] = "Unable to create account: Passwords don't match."
+      render :new
+    elsif params[:confirm_pass] == params[:password] && @user.save
+      flash[:notice] = "Your account has been created."
+      redirect_to "/profile"
+    else
+      flash[:error] = "Unable to create account: Required information missing."
+      redirect_to request.referer
```
Comment (Maxwell-Baird) [inline app/controllers/register_controller.rb]: Could we make redirect_to request.referer into render :new that way the code is more uniform and like the other parts of the if block?

File: app/controllers/register_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
+    if User.exists?(email: user_params[:email])
+      flash[:error] = "Email already in use. Try a different email."
+      render :new
+    elsif params[:confirm_pass] != params[:password]
+      flash[:error] = "Unable to create account: Passwords don't match."
+      render :new
```
Comment (colintalex) [inline app/controllers/register_controller.rb:13]: I like how we were able to keep this clean, for as complex as it is.

File: app/controllers/register_controller.rb (hunk 1 above)
Comment (colintalex) [inline app/controllers/register_controller.rb]: Let's look at possibly going back to render for these, just in the name of consistency

Comment (DavidTTran) [reply app/controllers/register_controller.rb]: done

File: app/views/register/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<h2>Register a New User</h2>
+
+<%= render partial: 'form', locals: { path: "/register",
+                                      method: :post,
+                                      button_name: "Submit",
+                                      user: params } %>
```
Comment (colintalex) [inline app/views/register/new.html.erb:6]: This was the magic touch!

File: db/migrate/20200407015715_remove_confirm_pass_from_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RemoveConfirmPassFromUsers < ActiveRecord::Migration[5.1]
+  def change
+    remove_column :users, :confirm_pass, :string
```
Comment (colintalex) [inline db/migrate/20200407015715_remove_confirm_pass_from_users.rb:3]: I agree with this, no need to have the confirmation in the DB

File: app/controllers/register_controller.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+class RegisterController < ApplicationController
+
+  def new
+  end
+
+  def create
```
Comment (DavidTTran) [inline app/controllers/register_controller.rb:6]: Feels weird to have User.new in the register controller but when we tried to put it in Users_controller it produced a template error.

**Review by colintalex — APPROVED** [review]
Looks good to me, kept the User controller since we'll be needing that down the road I presume

File: app/controllers/register_controller.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    @user = User.new(user_params)
+    if User.exists?(email: user_params[:email])
```
Comment (DavidTTran) [inline app/controllers/register_controller.rb:8]: This line can eventually be replaced with a `validate_uniqueness_of`

File: app/controllers/register_controller.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      flash[:error] = "Unable to create account: Passwords don't match."
+      render :new
+    elsif params[:confirm_pass] == params[:password] && @user.save
+      flash[:notice] = "Your account has been created."
+      redirect_to "/profile"
+    else
+      flash[:error] = "Unable to create account: Required information missing."
+      render :new
+    end
+  end
```
Comment (DavidTTran) [inline app/controllers/register_controller.rb:21]: Also in general, we're encouraged to keep controller methods short, but I don't see any way around this situation..

**Pull request merged**